### DISCUSSION TITLE
Downloaded 20 files but can no longer download from same website

### DISCUSSION THREAD

**Discussion by linzibrynn45** [aclap-dev/video-downloadhelper#973] [Help and Q&A]
### Your question

I downloaded 20 files (1 workout series) from Beachbodyondemand.com, no problem.  Waited a few days to try downloading more and now I get the **Download failed. Sadly we failed at downloading that specific media. We try to support as many websites as possible, so it would help a lot if you could report that error (it's anonymous!).**

I've cleared history/cache.  restarted computer.  logged out of beachbody and logged back in.  re-installed the VDH extension.  Tried downloading different programs.  Tried downloading the workout series that I already downloaded with VDH and it didn't work.  I moved all the 20 files from the dwhelper folder to my desktop too. Why isn't this working now??

### Addon & CoApp version and details

version: 9.0.2.12
target: microsoft
channel: stable
lang: en-US
coapp: {"found":true,"new_version":false,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19"}
license: {"accepted":true,"email":"<removed>","key":"700650c830464e52"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36 Edg/126.0.0.0
{
    "valid_license_message_has_been_displayed": true,
    "view_options": {
        "all_tabs": false,
        "hide_downloaded": false,
        "low_quality": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "sort_by_status": true,
        "sort_reverse": false
    },
    "open_count_store": 7
}

**Comment by GrampaWildWilly** [comment]
This sounds almost like a duplicate of #971.  You might want to read that one & try the suggestion there.  I don't use Edge so I have no idea if the latest beta is newer that the general availability version you are running.  But you can't know if it corrects your problem until you try it.

**Reply by linzibrynn45** [reply]
I changed my IP address, didn't work.  I downloaded the Edge Beta VDH and it didn't work.  I downloaded Firefox and VDH for Firefox and it didn't work.  Wondering if the website maybe blocked my username/password from VDH if that is even possible?  Any other ideas I can try?

**Reply by GrampaWildWilly** [reply]
Changing IP addresses is something that other user did because he believed the web site was blacklisting his IP address.  I question that.  He has not provided an answer to my question of how he determined his IP address was being blacklisted.  I suspect whatever was going wrong for him is unique to him.  Besides, switching IP addresses turned out not to be a solution for him, either.  At the time I mentioned that other problem, most of the dialog there had not yet occurred.  In light of what subsequently did get posted over there, I doubt my initial reaction of a duplicate problem is correct.

I am assuming your web site requires a paid subscription.  In that case, I can't visit any pages there myself.  (I repeat, as I like to do, that I do not work for VDH.  I'm just another user.)  I'm sure not signing up there just to try to figure this one out.  But you might be able to progress things.  Post a screenshot of one of your web pages using Firefox.  Specifically Firefox.  The information I'm looking for is provided by neither Chrome nor Edge.  Let's see a screenshot of one of your exercise pages in Firefox & we can take it from there.

**Reply by linzibrynn45** [reply]
Is this what you are looking for? 
[VDH.docx](https://github.com/user-attachments/files/16154173/VDH.docx)
I've tried downloading all of the file sizes too.

**Reply by GrampaWildWilly** [reply]
Yes.  Although the way I'm attaching the image here is more convenient.  I was hoping there would be a certain icon showing up in the area I've circled.

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/52a6b63c-9608-41c5-8c64-5ad40a2e520d)

But it's not there.  The icon would have indicated that the content is protected by DRM, which would mean downloading would be impossible.  But that's not the case so scratch that possibility.

**Comment by Mattmjs** [comment]
You should share the details about the video that fails, click the 3 dot button next to download. Click details and a page should
open. Share that here as text, not as an image.

**Reply by linzibrynn45** [reply]
Thanks.  I forgot how to do that from before. See below from Firefox. 


<html><body>

ab | 1 (incognito: false)
-- | --
LIIFT4 Program Overview: Start Your Journey Here
https://www.beachbodyondemand.com/programs/liift4/workouts#liift4-week-1-build-it with 10 headers
https://bod-cms-assets.prod.cd.beachbodyondemand.com/images/eonsdgcj/production/6839ebc04429cad29035c4e2c54a32a512bfafaf-1125x750.jpg
false
https://d197pzlrcwv1zr.cloudfront.net/LI4B1001/LI4B1001_Main.m3u8?Expires=1720750171&Key-Pair-Id=KLCJC6JAKT7X3&Signature=TfaLJQCIlJorh94po~YnExDjaJynCp50kNsHmdD16oxunkvLslwt1lKwcaPQlAIuG5hZB~rty8rSRICHTrPKpjcluOpuT9ourGNTObntT00n9VN3A0SyrhLSKwcUitTdZnpwZkzX4fkbKgRVBo-kR4~~MhZ8d5T78pJt8e05wD~eqGaS9eo~Ep45kdHvi2qG80DgQZMaZ2foWrm-88p9MTztJVjco0a4dBHYwbbd4T25B0DQAkJGcvSrT4XPMrvzBrTLtUofM5E97s5Iw2ifh23nhWfnKOcV5EYjECFtEC8rLQL4b3juzrHMZUbwCL29aWkE4zEx2U02tlcZPI6ZYQ__
https://d197pzlrcwv1zr.cloudfront.net/LI4B1001/LI4B1001_Main_B.m3u8
https://d197pzlrcwv1zr.cloudfront.net/LI4B1001/LI4B1001_Main_at2.m3u8
Mp4 b:Hls p:hls 2401s
AAC at unknown
H264 at 8597561, fps: 23.976, q:1080, {"width":1920,"height":1080}
https://d197pzlrcwv1zr.cloudfront.net/LI4B1001/LI4B1001_Main.m3u8?Expires=1720750171&Key-Pair-Id=KLCJC6JAKT7X3&Signature=TfaLJQCIlJorh94po~YnExDjaJynCp50kNsHmdD16oxunkvLslwt1lKwcaPQlAIuG5hZB~rty8rSRICHTrPKpjcluOpuT9ourGNTObntT00n9VN3A0SyrhLSKwcUitTdZnpwZkzX4fkbKgRVBo-kR4~~MhZ8d5T78pJt8e05wD~eqGaS9eo~Ep45kdHvi2qG80DgQZMaZ2foWrm-88p9MTztJVjco0a4dBHYwbbd4T25B0DQAkJGcvSrT4XPMrvzBrTLtUofM5E97s5Iw2ifh23nhWfnKOcV5EYjECFtEC8rLQL4b3juzrHMZUbwCL29aWkE4zEx2U02tlcZPI6ZYQ__
https://d197pzlrcwv1zr.cloudfront.net/LI4B1001/LI4B1001_Main_C.m3u8
https://d197pzlrcwv1zr.cloudfront.net/LI4B1001/LI4B1001_Main_at2.m3u8
Mp4 b:Hls p:hls 2401s
AAC at unknown
H264 at 6537835, fps: 23.976, q:1080, {"width":1920,"height":1080}
https://d197pzlrcwv1zr.cloudfront.net/LI4B1001/LI4B1001_Main.m3u8?Expires=1720750171&Key-Pair-Id=KLCJC6JAKT7X3&Signature=TfaLJQCIlJorh94po~YnExDjaJynCp50kNsHmdD16oxunkvLslwt1lKwcaPQlAIuG5hZB~rty8rSRICHTrPKpjcluOpuT9ourGNTObntT00n9VN3A0SyrhLSKwcUitTdZnpwZkzX4fkbKgRVBo-kR4~~MhZ8d5T78pJt8e05wD~eqGaS9eo~Ep45kdHvi2qG80DgQZMaZ2foWrm-88p9MTztJVjco0a4dBHYwbbd4T25B0DQAkJGcvSrT4XPMrvzBrTLtUofM5E97s5Iw2ifh23nhWfnKOcV5EYjECFtEC8rLQL4b3juzrHMZUbwCL29aWkE4zEx2U02tlcZPI6ZYQ__
https://d197pzlrcwv1zr.cloudfront.net/LI4B1001/LI4B1001_Main_D.m3u8
https://d197pzlrcwv1zr.cloudfront.net/LI4B1001/LI4B1001_Main_at2.m3u8
Mp4 b:Hls p:hls 2401s
AAC at unknown
H264 at 5130479, fps: 23.976, q:720, {"width":1280,"height":720}


</body>
</html>

**Reply by linzibrynn45** [reply]
This is from Edge:
<html>
<body>

ab | 974845450 (incognito: false)
-- | --
LIIFT4 Program Overview: Start Your Journey Here
https://www.beachbodyondemand.com/programs/liift4/workouts with 12 headers
https://bod-cms-assets.prod.cd.beachbodyondemand.com/images/eonsdgcj/production/6839ebc04429cad29035c4e2c54a32a512bfafaf-1125x750.jpg
true
https://bod-cms-assets.prod.cd.beachbodyondemand.com/files/eonsdgcj/production/e4ca2dc3fba2fcff2f2ccb9678884d25aea0e5f8.mp4
https://bod-cms-assets.prod.cd.beachbodyondemand.com/files/eonsdgcj/production/e4ca2dc3fba2fcff2f2ccb9678884d25aea0e5f8.mp4
none
Mp4 b:HTTPMedia p:non-adaptative unknowns
unknown at -1, fps: ?, q:?, no dim


</body>
</html>

**Reply by GrampaWildWilly** [reply]
I don't know why your data is here as just plain text instead of showing the URLs as clickable links.  When the VDH Hit Details page shows up, you should do Ctrl+a to highlight the full page, then Ctrl+Insert to copy it to the clipboard.  Then you should come over here to a post & do Shift+Insert to paste it in here.  When I do that, the result contains links you can click.  Well, whatever.  That's for future reference.  For this one, I extracted one URL from your post, like this:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/2ef03e14-7938-4af9-b995-dac5dac831ee)

That URL includes the string m3u8.  That means it's an HLS manifest.  With any luck, it would be an HLS master manifest.  And we're in luck.

[ffprobe.txt](https://github.com/user-attachments/files/16155243/ffprobe.txt)

There are errors in there that I can't explain.  But they don't seem to be stopping ffprobe from giving us information we can use.

At this point, this becomes so much like this other discussion #985 that I just worked on just a couple of hours ago.  It's uncanny how similar the two problems are.  The ffprobe report on your video shows there are English & Spanish audio tracks (the other problem had several more audio tracks) & there are English & Spanish subtitles (the other problem had a whole boatload of language captions).  But the solution is about the same.

[Exercise Video mp4 Log Filtered.txt](https://github.com/user-attachments/files/16155261/Exercise.Video.mp4.Log.Filtered.txt)
[Exercise Video vtt Log Filtered.txt](https://github.com/user-attachments/files/16155263/Exercise.Video.vtt.Log.Filtered.txt)

Now, unless you're willing to exercise your mind as hard as you're exercising your body, you're not going to get your videos the way I got this one:

![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/7539a4b1-836f-460b-ad65-4cfd6cdec6a6)

I put more hints about how to go about doing this over in #985.

Now, the question is, why didn't VDH recognize this?  #985 deals with YouTube which is just a can of worms.  But your web site is more or less standard.  I don't see any reason the multiple language audio tracks should trip up VDH.  In each Program, the first audio track is the English one.  There are multiple English audio tracks but just the one Spanish track repeated in each Program.  One possible complaint might be that you download a video & you get the Spanish audio when you want the English.  That might happen if the first audio track in a Program happens to be Spanish.  But here, the first audio track is always English, so I wouldn't expect that problem.  The 2 captions tracks, repeated in every Program, shouldn't be a problem either.  VDH is blind to subtitles . . . today.  In the future, that's supposedly not going to always be true but today, VDH doesn't handle captions.

Perhaps those errors in the ffprobe report are causing a problem for VDH.  I don't know what those errors are.  I'm kind of guessing that there's supposed to also be French subtitles with this but somebody bungled them.  That would make me guess that in addition, there's supposed to be a French audio track, but there isn't one.  In any case, I would hope VDH would ignore those errors the same way ffmpeg did.  501 not implemented.  Without actually seeing what that data is, there's no way of telling what they were trying to give us that ffmpeg doesn't know how to handle.

Bottom line.  I don't know why VDH isn't recognizing this content.  Did the web site change the way they're presenting the data?  I can't say since this is the first video from there that I've ever seen, so I have nothing to compare with.  Did they notify you, as a member of the site, that they were going to change something?  In any case, I've shown you a way you can get these videos even if it's a shade less convenient than VDH.  But you're going to have to learn some new things if you want to get this content right now.  Otherwise, you're at the mercy of however long it takes @paulrouget Paul to get around to putting a fix into VDH.

**Reply by Mattmjs** [reply]
> Bottom line. I don't know why VDH isn't recognizing this content.

I don't think that is the issue, @linzibrynn45 is getting detection but it is resulting in the download failed error message.
I used the same url shown in the image and pasted in a new tab page. It detected the video and clicked to download it.
It proceeded to download the video. No error message in the stable version for Firefox. I tried in the chrome version on
Ungoogled Chromium and it worked just like Firefox. The question is why is it failing for linzibrynn45.

**Reply by Mattmjs** [reply]
@linzibrynn45 , are using any proxy or browser vpn extensions ? Both of those could cause issues.
Also have you checked anti virus & firewall have not blocked companion app. The companion app comes with some
.exe files that are mistakenly blocked or put in quarantine.

**Reply by linzibrynn45** [reply]
Well, I just tried downloading it again and it worked!  I have no idea why, but if it stops working I will definitely use your suggestions above.  Thank you both, @GrampaWildWilly and @Mattmjs !!

**Reply by linzibrynn45** [reply]
Well, I was able to download 2 videos.  Tried a third and got the same download failed error message.  I don't have an anti virus (I know, bad) and I turned off my firewall.  Still got the same error message.  Turned firewall back on. Went to Extensions, I only have the VDH extension. No proxy or browser vpn extensions.  I've sort of come to the conclusion that VDH will work when it wants to which sucks because I need to cancel my Beachbody subscription by August 8th and was hoping to download a few more workout programs before I do. 

Also, it will let me download audio files.

**Reply by GrampaWildWilly** [reply]
Depends on how desperate you get.  There's always the ffmpeg method I showed you.

**Reply by linzibrynn45** [reply]
@GrampaWildWilly  how exactly do I try the ffmpeg? I'm pretty techy but can't figure it out from your posts above.

**Reply by GrampaWildWilly** [reply]
The rough outline goes like this:

You have to get the HLS master manifest.  Once you have that, most of the work is done.  The URL for that manifest is usually provided by the VDH Hit Details on the video.  You posted some of your Hit Details above.  The manifest we want is usually the one labelled `#0 main url` in the Hit Details.  You pass that URL into ffprobe.  Ffprobe is supplied with the CoApp, as is ffmpeg.  Generally, you'll find those on your Windows boot partition.  Do a `dir /s` to find them.  One of the files I attached above shows an execution of ffprobe.  The exact pattern of parameters I use is this:

```
ffprobe -protocol_whitelist file,crypto,data,http,https,tls,tcp -strict experimental -analyzeduration 25000000 -probesize 250000000 -allowed_extensions ALL "URL of HLS master manifest" 2>"ffprobe output file"
```

Once you run ffprobe, you need to analyze the output it generates.  Its output is in this general format:

```
Program 0
  Stream 0:0
  Stream 0:1
  possibly more Streams
Program 1
  Stream a:b
  Stream c:d
  possibly more Streams
Possibly more Programs
```

Each Program represents one variant of your video at a given resolution.  Each Stream represents one track of your video, typically video, audio, or captions.  You need to pay attention to the Stream identifiers, the things of the form `x:y` reported by ffprobe.  Those inform your subsequent invocation of ffmpeg.

I've attached a couple of logs from invocations of ffmpeg above.  Those logs include the exact commands I executed in each case.  The Stream identifiers from ffprobe go into the `-map` parameters of the ffmpeg commands.

There is no substitute for just plowing through the files I attached in my earlier post.  Those files show what happens in a lot of detail.  With a little practice, you'll get the hang of it.  There's a lot of detail that you can skim over & some detail that you need to pay attention to.  You say you're techy.  That's a refreshing change from the norm around here.  That means you'll figure it out quickly enough.  There is no manual for all of this.  I learned it by just studying the logs & seeing the patterns.  It took a while but it's not all that hard.

For a much more detailed explanation of the subject, refer to my tutorial in our old Google Groups forum:

https://groups.google.com/g/video-downloadhelper-q-and-a/c/oYDFx367MSY

There's also this:

https://groups.google.com/g/video-downloadhelper-q-and-a/c/sNfTCMYfiTU

It contains a number of examples of how to go about discovering the information you need to use ffmpeg.

Once you get the hang of this, it's pretty easy to repeat.  You'll want to write yourself some .bat files to hold the commands so you don't end up retyping those things over & over.

**Comment by tfood1** [comment]
A disk full condition will result in this error also. "This PC" in Windows Explorer will show how much space is available on all local disks.

**Reply by greaterthanstar** [reply]
> A disk full condition will result in this error also. "This PC" in Windows Explorer will show how much space is available on all local disks.

Oh my god man. Thank you for being the voice of every tech support person. "Is your computer turned on? Is your disk full?"

this solved my problem immediately hahahahahah. people (like me) forget that they go crazy once they install this and start dl'ing massive files. sheesh. I feel dumb. but thanks.
